backend/page_ops.py

```
@@ -268,34 +215,62 @@ def update_branch_page(
+                "page": _row_to_page(existing)
+            }
         
-        # Write back
-        page_path.write_text(json.dumps(page_data, indent=2, ensure_ascii=False), encoding='utf-8')
-        logger.info(f"Updated page for branch '{branch_name}' in {repo_name}")
+        # Add WHERE clause params
+        params.extend([user_id, repo_id, branch_name])
+        
+        await database.execute(
+            f"""UPDATE branch_pages SET {', '.join(updates)}
+                WHERE user_id = %s AND repo_id = %s AND branch_name = %s""",
```

Ruff (0.14.10)

Possible SQL injection vector through string-based query construction

(S608)